src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1128,14 +1128,19 @@ void Unit::CalculateMeleeDamage(Unit* victim, CalcDamageInfo* damageInfo, Weapon
 
     for (uint8 i = 0; i < MAX_ITEM_PROTO_DAMAGES; ++i)
     {
+        // only players have secondary weapon damage
```

Comment:
ain't some creatures supposed to use offhand weapon?

Reply:
This is not offhand calculation, this is the secondary effects of each damage types

Reply:
F.e. http://db.logon3.com/?item=1482 
+4 - 8 Shadow Damage
This is secondary weapon damage